src/util/orderUtils.ts

```diff
@@ -0,0 +1,29 @@
+/*
+  Copyright 2018 Set Labs Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+'use strict';
+
+import { BigNumber } from '.';
+
+export const calculatePartialAmount = (
```

Comment:
💯 on the idea.

`calculatePartialAmount` is more of a general math utility than order-specific. Perhaps it belongs in `bignumber.ts` or `commonMath.ts`